src/initiate.js

```diff
@@ -116,8 +142,31 @@ export const initiateRendering = async (
       outputDir,
       outputFilename,
     );
+
+    console.log(
+      `\x1b[32m${outputFilename}.mbtiles has been successfully generated!\x1b[0m`,
+    );
+
+    // if successful, return the metadata
+    return {
+      style: style,
+      status: metadata.status,
+      errorCode: metadata.errorCode,
+      errorMessage: metadata.errorMessage,
+      filename: metadata.filename,
+      filesize: metadata.filesize,
+      numberOfTiles: metadata.numberOfTiles,
+      numberOfAttempts: 1,
```

Comment:
I'm thinking your code does not need to handle retries explicitly; rather the queuing system will retry to deliver the same message if it hadn't succeeded (been deleted from the queue by the task worker) within the lease time.

All your task worker code has to do is read [`DequeuedMessageItem.dequeueCount`](https://learn.microsoft.com/en-us/javascript/api/@azure/storage-queue/dequeuedmessageitem?view=azure-node-latest):

```js
const response = await sourceQueueClient.receiveMessages({
      visibilityTimeout: 2 * 60 * 60,
});
const message = response.receivedMessageItems[0];
const numberOfAttempts = message.dequeueCount;
```